### PR TITLE
app-crypt/moolticute: Version bump to 1.00.1 and fixes for udev rules

### PR DESCRIPTION
**Pull request by CasInvictus**
Provides a version bump to the latest stable version of moolticute, and dropped support for the older v0.43.19. This also fixes a few issues with udev the previous builds had.

Closes: https://bugs.gentoo.org/854249
Signed-off-by: Casper Hanegreefs <casinvictusgentoo@proton.me>

### CODE REVIEW

File: app-crypt/moolticute/files/moolticuted.init [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #!/sbin/openrc-run
-# Copyright 1999-2017 Gentoo Foundation
+# Copyright 1999-2021 Gentoo Authors
```
Comment (scardracs) [inline app-crypt/moolticute/files/moolticuted.init:2]: 2023

Comment (CasInvictus) [reply app-crypt/moolticute/files/moolticuted.init]: Looks like i missed one. Should be fixed now

**Comment by scardracs** [comment]
Runiing it on a lxd container. I'll report to you if everything is ok :)

**Comment by scardracs** [comment]
Link for 1.00.1 is broken, it should be `SRC_URI=https://github.com/mooltipass/moolticute/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz`

**Comment by scardracs** [comment]
```
 * VDB: detected possibly incorrect RDEPEND (app-crypt/moolticute-0.43.19-r1)
 * dev-libs/libusb <  # these 2 can be ignored IMHO
 * dev-qt/qttest:5 < 
 *                 > sys-apps/systemd:=
```

**Comment by CasInvictus** [comment]
> ```
>  * VDB: detected possibly incorrect RDEPEND (app-crypt/moolticute-0.43.19-r1)
>  * dev-libs/libusb <  # comment: these 2 can be ignored IMHO
>  * dev-qt/qttest:5 < 
>  *                 > sys-apps/systemd:=
> ```

Alright thank you. I'll be running tests with the updated SRC_URI and removed libusb and qttest RDEPEND, but is the systemd depend necessary here?

**Comment by thesamesam** [comment]
Things on the left **need to be checked** as they're simply not linked against. They might still be e.g. dlopened.

Things on the right _are_ depended on in some way (linked against) so you need to adjust the ebuild accordingly. Build system toggles may exist.

**Comment by scardracs** [comment]
> > ```
> >  * VDB: detected possibly incorrect RDEPEND (app-crypt/moolticute-0.43.19-r1)
> >  * dev-libs/libusb <  # comment: these 2 can be ignored IMHO
> >  * dev-qt/qttest:5 < 
> >  *                 > sys-apps/systemd:=
> > ```
> 
> 
> Alright thank you. I'll be running tests with the updated SRC_URI and removed libusb and qttest RDEPEND, but is the systemd depend necessary here?

I don’t think removing these are necessary. Probably the systemd request is because I’m running lxd in a systemd profile

**Comment by CasInvictus** [comment]
Okay thanks for the info! I'm still learning more about the ebuilds :D

I will commit the SRC_URI fix already, and check the RDEPEND stuff later!

**Comment by CasInvictus** [comment]
Sorry about the delay, but I have checked the dependencies now! Libusb is of course needed for connecting the mooltipass device via USB, so it should be left in. Qttest is referenced in some files (/tests/ directory) which are in the upstream code, so it is probably better to leave it in here.

For systemd I was a bit unsure on how to continue, but i was thinking the following: (correct me if this is a bad idea :P)
Add a "inherit systemd" and in src_install a "systemd_dounit systemd/moolticuted.service" (as the service file is provided by upstream, but not in the ebuild yet I found out). When adding this, does it mean it would also need a systemd dependency?

**Comment by ceamac** [comment]
Yes, but conditional (depends on USE `systemd`)

**Comment by CasInvictus** [comment]
Alright I added the conditional systemd dependency (with USE flag), and the service file to the ebuild

File: app-crypt/moolticute/moolticute-0.43.19-r1.ebuild [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 	EGIT_REPO_URI="https://github.com/mooltipass/moolticute.git"
 	inherit git-r3
 else
-	SRC_URI="https://github.com/mooltipass/moolticute/archive/v${PV/_/-}.tar.gz -> ${P}.tar.gz"
+	SRC_URI="https://github.com/mooltipass/moolticute/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
 	KEYWORDS="~amd64 ~arm"
 fi
 
-inherit xdg-utils qmake-utils udev
+inherit systemd xdg-utils qmake-utils udev
```
Comment (ceamac) [inline app-crypt/moolticute/moolticute-0.43.19-r1.ebuild:14]: Please sort the eclasses alphabetically.  Also move this line above if [[ ...

If you look into `skel.ebuild`, inherit should come right after `EAPI`.

File: app-crypt/moolticute/moolticute-0.43.19-r1.ebuild [hunk 3]
```diff
@@ -29,6 +29,7 @@ RDEPEND="
 	dev-qt/qttest:5
 	dev-qt/qtwebsockets:5
 	dev-qt/qtwidgets:5
+	systemd? ( sys-apps/systemd )
```
Comment (ceamac) [inline app-crypt/moolticute/moolticute-0.43.19-r1.ebuild:32]: for `!systemd` you need to depend on `sys-apps/systemd-utils`

**Review by ceamac — CHANGES_REQUESTED** [review]
You should revbump 0.43.19 - you changed a RDEPEND, also changed installed files, and adding a `systemd` USE flag will not force a rebuild on openrc systems.

You should also squash the commits - maybe one commit for 0.43.19 and another for the others, since the ebuilds are otherwise identical - which is a good thing.

Thank you for working on this!

**Comment by ceamac** [comment]
You can also remove the `S=...` line, it's not needed

**Comment by CasInvictus** [comment]
So I added the changes into a different commit per version. As I previously did quite some commits with multiple file edits, it was easier to just do a reset and redo the commits (the reason for the force-push). But I think I did all the mentioned fixes

**Comment by CasInvictus** [comment]
If there are still issues, let me know :)

**Review by ceamac — APPROVED** [review]
Great!  Thank you.

**Pull request closed without merging**